### PR TITLE
Moving cloudtrail (2) resources to v1beta1 version

### LINKED ISSUE #35: Moving cloudtrail (2) resources to v1beta1 version
Moving 2 resources in the `cloudtrail` group:
- [x] [`aws_cloudtrail`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail)
- [x] [`aws_cloudtrail_event_data_store`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail_event_data_store)

Extracted from https://github.com/upbound/official-providers/issues/248

### PR DESCRIPTION
**Pull request by dverveiko**
### Description of your changes
Add resources:
- [x] [`aws_cloudtrail`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail)
- [x] [`aws_cloudtrail_event_data_store`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudtrail_event_data_store)

Fixes #35 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
`aws_cloudtrail` - Manually and [Uptest](https://github.com/upbound/provider-aws/actions/runs/3636755173)
`aws_cloudtrail_event_data_store` - Manually and [Uptest](https://github.com/upbound/provider-aws/actions/runs/3636758425)

### CODE REVIEW

**Comment by dverveiko** [comment]
/test-examples="examples/cloudtrail/trail.yaml"

**Comment by dverveiko** [comment]
/test-examples="examples/cloudtrail/eventdatastore.yaml"

File: examples/cloudtrail/trail.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        testing.upbound.io/example-name: foo
+    s3KeyPrefix: prefix
+
+---
+
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/external-name: op-rgszzcuww
```
Comment (ezgidemirel) [inline examples/cloudtrail/trail.yaml:24]: would be good to use a random external-name in here too

```suggestion
    crossplane.io/external-name: ${Rand.RFC1123Subdomain}
```

**Review by ezgidemirel — COMMENTED** [review]
Thanks @dverveiko! Just left a small comment

File: examples/cloudtrail/trail.yaml (hunk 1 above)
Comment (ezgidemirel) [reply examples/cloudtrail/trail.yaml:24]: As we discussed offline, we need that name in the BucketPolicy resource. So we can keep it hard-coded.

**Review by ezgidemirel — APPROVED** [review]

**Pull request merged**